Sources/SwiftSyntax/Diagnostic.swift

```diff
@@ -16,14 +16,14 @@
 /// diagnostic.
 public enum FixIt: Codable {
   /// Remove the characters from the source file over the provided source range.
-  case remove(SourceRange)
+  case remove(ByteSourceRange)
 
   /// Insert, at the provided source location, the provided string.
-  case insert(SourceLocation, String)
+  case insert(Int, String)
```

Comment:
Have you considered using `AbsolutePosition` instead of `Int`, in the places where you replace `SourceLocation` with an offset ?

Reply:
I did. I would feel more inclined to using it if `ByteSourceRange` uses `AbsolutePosition` underneath, but it currently doesn't.

Reply:
`ByteSourceRange` can be used to represent a relative range so not quite sure if `AbsolutePosition` is proper for it.

But since `Syntax` nodes have a `position` property, I think it would be useful to also have a `Diagnostic.position` property, using the same type as the `Syntax` nodes so they can be easily compared.